From 6d4656ddd323a5734d1285732bde5b743fcaeab7 Mon Sep 17 00:00:00 2001
From: Alain Volmat <alain.volmat@foss.st.com>
Date: Sat, 2 Dec 2023 22:38:45 +0100
Subject: [PATCH 0806/1141] media: dcmipp: pixelcap: add all fields in default
 format

Set all field of the v4l2_mbus_format structure for the default format.

Change-Id: I5d2aa29b3c3cb827e8cc0b6874602e645b69c943
Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/344614
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Philippe CORNU <philippe.cornu@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
---
 .../media/platform/st/stm32/stm32-dcmipp/dcmipp-pixelcap.c   | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-pixelcap.c b/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-pixelcap.c
index d37104d8ce1e..e99598d1dc63 100644
--- a/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-pixelcap.c
+++ b/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-pixelcap.c
@@ -202,7 +202,10 @@ static const struct v4l2_pix_format fmt_default = {
 	.height = DCMIPP_FMT_HEIGHT_DEFAULT,
 	.pixelformat = V4L2_PIX_FMT_RGB565,
 	.field = V4L2_FIELD_NONE,
-	.colorspace = V4L2_COLORSPACE_DEFAULT,
+	.colorspace = V4L2_COLORSPACE_REC709,
+	.ycbcr_enc = V4L2_YCBCR_ENC_DEFAULT,
+	.quantization = V4L2_QUANTIZATION_DEFAULT,
+	.xfer_func = V4L2_XFER_FUNC_DEFAULT,
 };
 
 struct dcmipp_pixelcap_buffer {
-- 
2.39.2

